Formula/pyroscope.rb

```diff
@@ -0,0 +1,104 @@
+class Pyroscope < Formula
+  desc "Open source continuous profiling software"
+  homepage "https://pyroscope.io"
+  url "https://dl.pyroscope.io/release/pyroscope-0.5.1-source.tar.gz"
+  sha256 "a5f138a04a7a3b31c9e693370d9c214e3a6724fe14fbe123f8b7adc4ae28aff3"
+  license "Apache-2.0"
+  head "https://github.com/pyroscope-io/pyroscope.git", branch: "main"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "rust" => :build
+  depends_on "yarn" => :build
+  depends_on "zstd" => :build
+
+  on_linux do
+    depends_on "libunwind"
+  end
+
+  def install
+    system "make", "build-rust-dependencies"
+    system "make", "install-build-web-dependencies"
+    system "make", "build-release"
+
+    bin.install "bin/pyroscope"
+  end
+
+  def post_install
+    (var/"log/pyroscope").mkpath
+    (var/"lib/pyroscope").mkpath
+    (etc/"pyroscope").mkpath
+
+    (etc/"pyroscope/server.yml").write pyroscope_conf unless File.exist?((etc/"pyroscope/server.yml"))
+  end
+
+  def pyroscope_conf
+    <<~EOS
+      ---
+      storage-path: #{var}/lib/pyroscope
+    EOS
+  end
+
+  plist_options manual: "pyroscope server -config #{HOMEBREW_PREFIX}/etc/pyroscope/server.yml"
+
+  def plist
+    <<~EOS
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+      <plist version="1.0">
+        <dict>
+          <key>KeepAlive</key>
+          <dict>
+            <key>SuccessfulExit</key>
+            <false/>
+          </dict>
+          <key>Label</key>
+          <string>#{plist_name}</string>
+          <key>ProgramArguments</key>
+          <array>
+            <string>#{bin}/pyroscope</string>
+            <string>server</string>
+            <string>-config</string>
+            <string>#{etc}/pyroscope/server.yml</string>
+          </array>
+          <key>RunAtLoad</key>
+          <true/>
+          <key>WorkingDirectory</key>
+          <string>#{var}/lib/pyroscope</string>
+          <key>StandardErrorPath</key>
+          <string>#{var}/log/pyroscope/server-stderr.log</string>
+          <key>StandardOutPath</key>
+          <string>#{var}/log/pyroscope/server-stdout.log</string>
+          <key>SoftResourceLimits</key>
+          <dict>
+            <key>NumberOfFiles</key>
+            <integer>10240</integer>
+          </dict>
+        </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    require "pty"
+    require "timeout"
+
+    # first test
+    system bin/"pyroscope", "--version"
+
+    # avoid stepping on anything that may be present in this directory
+    tdir = File.join(Dir.pwd, "pyroscope-test")
+    Dir.mkdir(tdir)
+
+    r, w, pid = PTY.spawn(bin/"pyroscope", "-api-bind-addr :50100")
```

Comment:
Does this actually need a PTY? Or is `fork { exec "cmd" }` enough?

Reply:
I tried using `fork { exec "cmd" }` and it shows only `pid`. In that case, I'm validating if the string `"starting HTTP server"` was found when the server starts.